src/util/async/impl/Any.hpp

```diff
@@ -0,0 +1,51 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <any>
+#include <type_traits>
+
+namespace util::async::impl {
+
+/**
+ * @brief A wrapper for std::any to workaround issues with boost.outcome
+ */
+class Any {
+    std::any value_;
+
+public:
+    Any() = default;
+    Any(Any const&) = default;
+    Any(Any&&) = default;
+
+    // note: this needs to be `auto` instead of `std::any` because of a bug in gcc 11.4
```

Comment:
No, this is just another bug. The Any wrapper is needed because `util::Expected` is using Boost.Outcome and there is an issue using it with `std::any` directly. `std::expected` from C++23 does not have this problem so we will be able to remove this wrapper in favour of `std::any` as well as remove `util::Expected` in favour of `std::expected`. 

Created issues: #1173 #1174 
Will add a note and reference the issues as comments to avoid future confusion